tests/main/interfaces-network-control/task.yaml

```diff
@@ -20,31 +20,34 @@ environment:
     ARP_ENTRY_ADDR: "30.30.30.30"
 
 prepare: |
+    #shellcheck source=tests/lib/snaps.sh
     . "$TESTSLIB/snaps.sh"
+    #shellcheck source=tests/lib/systemd.sh
     . "$TESTSLIB/systemd.sh"
     echo "Given a snap declaring a plug on the network-control interface is installed"
     install_local network-control-consumer
 
     echo "And a network service is up"
-    printf "#!/bin/sh -e\nwhile true; do echo \"HTTP/1.1 200 OK\n\nok\n\" |  nc -l -p $PORT; done" > $SERVICE_FILE
-    chmod a+x $SERVICE_FILE
-    systemd_create_and_start_unit $SERVICE_NAME "$(readlink -f $SERVICE_FILE)"
     # shellcheck source=tests/lib/network.sh
-    . $TESTSLIB/network.sh
-    wait_listen_port "$PORT"
+    . "$TESTSLIB"/network.sh
+    make_network_service "$SERVICE_NAME" "$SERVICE_FILE" "$PORT"
 
 restore: |
+    #shellcheck source=tests/lib/systemd.sh
     . "$TESTSLIB/systemd.sh"
+    #shellcheck source=tests/lib/network.sh
     . "$TESTSLIB/network.sh"
 
-    systemd_stop_and_destroy_unit $SERVICE_NAME
-    rm -f *.output $SERVICE_FILE
-    arp -d $ARP_ENTRY_ADDR -i $(get_default_iface) || true
+    #shellcheck disable=SC2153
```

Comment:
what was tripping it up here?

Reply:
`SERVICE_NAME` which is only defined in `environment` section and injected by spread.